### ISSUE TITLE
Looks like I can't compile the plugin.

### ISSUE THREAD

**Issue by samuelh2005** [lokka30/ElementalEconomy#7]
# Introduction
I've done a simple `mvn package` and `mvn compile`  and get the error below. I'm not sure if I have done anything else wrong.

## Versions
* Java version
```
$ java --version
openjdk 17.0.1 2021-10-19
IBM Semeru Runtime Open Edition 17.0.1.0 (build 17.0.1+12)
Eclipse OpenJ9 VM 17.0.1.0 (build openj9-0.29.1, JRE 17 Windows 10 amd64-64-Bit Compressed References 20211207_55 (JIT enabled, AOT enabled)
OpenJ9   - 7d055dfcb
OMR      - e30892e2b
JCL      - fc67fbe50a0 based on jdk-17.0.1+12)
```

* Maven version
```
$ mvn -vesrion
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: E:\java\apache-maven-3.8.6
Java version: 17.0.1, vendor: IBM Corporation, runtime: C:\Users\samro\.jdks\semeru-17.0.1
Default locale: en_GB, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

* Git version
```
$ git --version
git version 2.41.0.windows.3
```

## The error
```
$ mvn package
[WARNING]
[WARNING] Some problems were encountered while building the effective settings
[WARNING] Expected root element 'settings' but found 'servers' (position: START_TAG seen <servers>... @1:9)  @ C:\Users\samro\.m2\settings.xml, line 1, column 9
[WARNING]
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] Polyconomy                                                         [pom]
[INFO] Polyconomy Plugin for Bukkit                                       [jar]
[INFO]
[INFO] -----------< io.github.arcaneplugins.polyconomy:polyconomy >------------
[INFO] Building Polyconomy 0                                              [1/2]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] ----------< io.github.arcaneplugins.polyconomy:plugin-bukkit >----------
[INFO] Building Polyconomy Plugin for Bukkit 0.1.0-alpha-0                [2/2]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ plugin-bukkit ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO]
[INFO] --- kotlin-maven-plugin:1.7.21:compile (compile) @ plugin-bukkit ---
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt: (731, 27) Unresolved reference: create
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt: (732, 31) Unresolved reference: Type
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt: (803, 27) Unresolved reference: create
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt: (804, 31) Unresolved reference: Type
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Polyconomy 0 ....................................... SUCCESS [  0.002 s]
[INFO] Polyconomy Plugin for Bukkit 0.1.0-alpha-0 ......... FAILURE [  9.937 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.011 s
[INFO] Finished at: 2023-08-12T13:42:41+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.7.21:compile (compile) on project plugin-bukkit: Compilation failure: Compilation failure:
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt:[731,27] Unresolved reference: create
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt:[732,31] Unresolved reference: Type
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt:[803,27] Unresolved reference: create
[ERROR] D:\minecraft\plugins\Polyconomy\plugin-bukkit\src\main\kotlin\io\github\arcaneplugins\polyconomy\plugin\bukkit\economy\storage\impl\local\YamlStorageHandler.kt:[804,31] Unresolved reference: Type
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :plugin-bukkit
```

**Comment by lokka30** [comment]
Hey @ajh123, this is caused since Polyconomy's codebase hasn't quite finished in updating to Treasury v2 and uses some outdated references :)

**Comment by samuelh2005** [comment]
> Hey @ajh123, this is caused since Polyconomy's codebase hasn't quite finished in updating to Treasury v2 and uses some outdated references :)

@lokka30 When do you think this might be fixed?

**Comment by lokka30** [comment]
> > Hey @ajh123, this is caused since Polyconomy's codebase hasn't quite finished in updating to Treasury v2 and uses some outdated references :)
> 
> @lokka30 When do you think this might be fixed?

Unfortunately I have no time to spare for plugin development at the moment besides replying to user discussions and issues. Full time uni + work + routine volunteering is a bad combination 😅

This is the first thing I will tackle once I'm back to working on Polyconomy.

**Comment by lokka30** [comment]
Very late to resolve this! Fixed in 7fb63765ffe22dca2cfe1d599cd69883a56422b7

**Comment by lokka30** [comment]
Hi @ajh123, Polyconomy now has working releases and is stable enough, calling it 'beta' since there's still [some more features](https://github.com/orgs/ArcanePlugins/projects/10/views/1) to work on (and it needs to be out in the wild for a bit to learn about any possible hidden bugs).
